src/utilities/createRouterRoutes.ts

```diff
@@ -10,13 +10,17 @@ type ParentContext = {
   parentDepth?: number,
 }
 
-export function createRouterRoutes(routes: Routes, parentContext: ParentContext = {}): RouterRoute[] {
+export function createRouterRoutes<T extends Routes>(routes: T, parentContext: ParentContext = {}): RouterRoute[] {
   const { parentPath = [], parentQuery = [], parentMatches = [], parentDepth = 0 } = { ...parentContext }
 
   return routes.reduce<RouterRoute[]>((value, route) => {
     const path = typeof route.path === 'string' ? createPath(route.path, {}) : route.path
     const query = typeof route.query === 'string' ? createQuery(route.query, {}) : route.query ?? { query: '', params: {} }
 
+    if (hasDuplicateParams(path, query)) {
+      throw new DuplicateParamsError()
+    }
```

Comment:
calling `path` and `query` will also throw this type of error if there are duplicates WITHIN the path or query. This line ensures that if path has `foo` and query also has `foo` we get an exception there too

Reply:
You mentioned path and query will throw errors. But I'm not finding that in this PR. Is that a future plan or what are you thinking here. Possible I just missed it but I looked lol.

Reply:
both `path` and `query` call `getParamsForString` to populate params. If you look in that utility, you'll see the following

```ts
const paramPattern = /:\??([\w]+)(?=\W|$)/g
const matches = Array.from(string.matchAll(paramPattern))

return matches.reduce<Record<string, Param>>((value, [match, paramName]) => {
  const isOptional = match.startsWith(':?')
  const param = getParam(params, paramName)

  if (paramName in value) {
    throw new DuplicateParamsError(paramName)
  }

  value[paramName] = isOptional ? optional(param) : param

  return value
```

Reply:
ah gotcha 👍  ty